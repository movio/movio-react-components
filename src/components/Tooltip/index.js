// @flow
import React from 'react';
import classnames from 'classnames';

import styles from './tooltip.css';

type Props = {
  className?: string,
  children?: ReactChildren,
};

const Tooltip = ({ className, children }: Props) => (
  <div className={classnames(className, styles.container)}>
    {children}
  </div>
);

export default Tooltip;
export { styles };
