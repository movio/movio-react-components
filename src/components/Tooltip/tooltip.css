@import "variables.css";

:root {
  --color-bg: var(--color-neutral-2);
  --color-shadow: rgba(0, 0, 0, 0.3);
}

.container {
  padding: 0.15rem 0.5rem;
  margin: 0.5rem;
  background: var(--color-bg);
  font-size: 1rem;
  line-height: 2rem;
  color: white;
  border-radius: var(--border-radius);
  border: 2px solid white;
  box-shadow: 0 2px 4px var(--color-shadow);
  z-index: 9;
}

