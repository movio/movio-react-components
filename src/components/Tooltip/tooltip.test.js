import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Tooltip, { styles } from './index';

describe('<Tooltip />', () => {
  it('should correctly render children, when text', () => {
    const component = <Tooltip>Title</Tooltip>;
    const wrapper = shallow(component);
    expect(wrapper.text()).toEqual('Title');
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('tooltip-basic');
  });

  it('should append a className to the default, if provided', () => {
    const component = <Tooltip className="test-class">Title Tooltip</Tooltip>;
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(`test-class ${styles.container}`);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('tooltip-custom-classname');
  });

  it('should use the default className from the stylesheet, if not provided', () => {
    const component = <Tooltip>Title Tooltip</Tooltip>;
    const wrapper = shallow(component);
    expect(wrapper.prop('className')).toEqual(styles.container);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot('tooltip-default-classname');
  });
});

