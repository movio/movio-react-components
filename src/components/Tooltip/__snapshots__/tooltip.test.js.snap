// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tooltip-basic 1`] = `
<div
  className="container"
>
  Title
</div>
`;

exports[`tooltip-custom-classname 1`] = `
<div
  className="test-class container"
>
  Title Tooltip
</div>
`;

exports[`tooltip-default-classname 1`] = `
<div
  className="container"
>
  Title Tooltip
</div>
`;
